### DISCUSSION TITLE
Emit seq metrics in prometheus format at https://<seq-url>/metrics

### DISCUSSION THREAD

**Discussion by agileramblings** [datalust/seq-tickets#1639] [Ideas]
Hello peeps,

There is a dockerhub image [here](https://hub.docker.com/r/smokserwis/seq-prometheus-metrics) that performs this task, but would this be something that seq could do natively?  

Cheers,
Dave

**Comment by nblumhardt** [comment]
Hi Dave! Thanks for the note. This seems worth exploring!

Are you mostly interested in Seq's internal diagnostic metrics (i.e. how Seq is performing, as per the linked container), or metrics computed from user data logged to Seq?

**Reply by agileramblings** [reply]
I was originally thinking of the ingestion and storage metrics (and any other internal Seq performance metrics). An extension mechanism that allowed users to add their own Prometheus metrics to the /metrics endpoint would be interesting as well but lower priority. 

We are planning to use Seq for all log ingestion, Prometheus for all metrics, and Alertmanager for alerting from Prometheus metrics. It would be great to get alerts if Seq was running into troubles, or unavailable for any reason.

**Reply by nblumhardt** [reply]
Thanks for the follow-up, Dave :+1:
